src/pages/index.js

```diff
@@ -90,19 +187,88 @@ function openEditCardModal() {
   addCardModal.open();
 }
 
+function openProfileImageModal() {
+  editAvatarModal.open();
+}
+
 /*------------------------------------------------------------------*/
-/*                          Event Objects                            */
+/*                          EVENT HANDLERS                            */
 /*------------------------------------------------------------------*/
 
 function handleProfileEditSubmit({ title, description }) {
-  userInfo.setUserInfo({ title, description });
-  profileEditModal.close();
+  profileEditModal.setLoading(true);
+  api.updateProfileInfo({ title, description }).then((res) => {
+    profileUserInfo.setUserInfo({ title, description });
+    profileEditModal.setLoading(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.